### ISSUE TITLE
Graalvm23 support

### ISSUE THREAD

**Issue by Oregano1963** [FabricMC/fabric#4099]
On a fresh 1.21.1 install of fabric api 104.0 on fabric 0.16.5, the game crashes when using GraalVM 23.

Graal Download: https://www.graalvm.org/downloads/

logs: https://mclo.gs/xChF8DR

**Comment by apple502j** [comment]
Use Java 21 or 22.

**Comment by modmuss50** [comment]
No, this is a valid issue, we should support Java 23 unfortunately we are blocked waiting on an ASM update. The latest released version that adds support for Java 23 has a bug preventing us from being able to update. I will re-open this issue and move it to the correct repo.
